site/cds_rdm/tasks.py

```diff
@@ -140,3 +147,224 @@ def sync_users():
+            updated_name["affiliations"].append({"name": user_affiliation})
+            updated = True
+
+        user_orcid = user.user_profile.get("orcid")
+        name_orcids = [
+            identifier["identifier"]
+            for identifier in name.get("identifiers", [])
+            if identifier["scheme"] == "orcid"
+        ]
+        if user_orcid and user_orcid not in name_orcids:
+            if "identifiers" not in updated_name:
+                updated_name["identifiers"] = []
+            updated_name["identifiers"].append(
+                {"scheme": "orcid", "identifier": user_orcid}
+            )
+            updated = True
+
+        for prop in prop_values:
+            user_prop = user.user_profile.get(prop)
+            name_prop = name.get("props", {}).get(prop)
+            if user_prop and user_prop != name_prop:
+                if "props" not in updated_name:
+                    updated_name["props"] = {}
+                updated_name["props"][prop] = user_prop
+                updated = True
+
+        # add the email as props
+        if user.email and user.email != name.get("props", {}).get("email"):
+            if "props" not in updated_name:
+                updated_name["props"] = {}
+            updated_name["props"]["email"] = user.email
+            updated = True
+
+        return updated_name if updated else None
+
+    def _update_name(user, name_dict, is_orcid=False):
+        """Updates the name with the new values."""
+        updated_name = _check_if_update_needed(user, name_dict, is_orcid)
+        if updated_name:
+            try:
+                service.update(system_identity, name_dict.get("id"), updated_name)
+            except ValidationError as e:
+                current_app.logger.error(f"Error updating name for user {user.id}: {e}")
+
+    def _create_new_name(user):
+        name = {"id": str(user.id), "props": {"is_cern": True}}
+
+        if user.user_profile.get("given_name"):
+            name["given_name"] = user.user_profile.get("given_name")
+        if user.user_profile.get("family_name"):
+            name["family_name"] = user.user_profile.get("family_name")
+        if user.user_profile.get("affiliations"):
+            name["affiliations"] = [{"name": user.user_profile.get("affiliations", "")}]
+        if user.user_profile.get("orcid"):
+            name["identifiers"] = [
+                {"scheme": "orcid", "identifier": user.user_profile.get("orcid")}
+            ]
+        for prop in prop_values:
+            if user.user_profile.get(prop):
+                name["props"][prop] = user.user_profile.get(prop)
+        try:
+            service.create(system_identity, name)
+        except ValidationError as e:
+            current_app.logger.error(f"Error creating name for user {user.id}: {e}")
+
+    users = []
+    # Allows to sync a single user
+    if user_id:
+        user = User.query.get(user_id)
+        users = [user] if user else []
```

Comment:
I think
```suggestion
        users = [user]
```
won't `.get` raise an exeption if user is not found for the given id?